drivers/net/native_sim_sockets.c

```diff
@@ -0,0 +1,707 @@
+/**
+ * Copyright (c) 2023 Marcin Niestroj
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/net/ethernet.h>
+#include <zephyr/net/net_ip.h>
+#include <zephyr/net/offloaded_netdev.h>
+#include <zephyr/net/socket_offload.h>
+#include <zephyr/posix/fcntl.h>
+#include <zephyr/sys/fdtable.h>
+
+#include "sockets_internal.h"
+#include "native_sim.h"
+#include "native_sim_dns.h"
+#include "native_sim_errno.h"
+#include "native_sim_fcntl.h"
+
+/* Increment by 1 to make sure we do not store the value of 0, which has
+ * a special meaning in the fdtable subsys.
+ */
+#define FD_TO_OBJ(fd) ((void *)(intptr_t)((fd) + 1))
+#define OBJ_TO_FD(obj) (((int)(intptr_t)(obj)) - 1)
+
+static int socket_family_to_nsim(int family, int *family_nsim)
+{
+	switch (family) {
+	case AF_UNSPEC:
+		*family_nsim = NSIM_AF_UNSPEC;
+		break;
+	case AF_INET:
+		*family_nsim = NSIM_AF_INET;
+		break;
+	case AF_INET6:
+		*family_nsim = NSIM_AF_INET6;
+		break;
+	default:
+		return -NSIM_EAFNOSUPPORT;
```

Comment:
Same comment as for _adapt about these error printing something to guide future developers.

Reply:
I have not added warnings here, since this function is used by `nsos_is_supported()` in order to figure out whether this driver is able to handle specific socket. If not (which is possible for example for TLS sockets), then there would be warning log generated even in a perfectly valid flow.

Also the error codes returned here are very descriptive, i.e. it is clear that either protocol is not supported or socket family is not supported.